engine/gamesys/src/gamesys/scripts/script_physics.cpp

```diff
@@ -1191,8 +1193,15 @@ namespace dmGameSystem
         void* comp_world = 0x0;
         GetCollisionObject(L, 1, collection, &comp, &comp_world);
         
-        dmhash_t group_hash = dmGameSystem::GetCollisionGroup(comp_world, comp);
-        dmScript::PushHash(L, group_hash);
+        dmhash_t group_hash;
+        dmGameSystem::GetCollisionGroup(comp_world, comp, group_hash);
+        if (!group_hash) {
+            dmhash_t url = dmScript::CheckHashOrString(L, 1);
+            dmLogWarning("No collision group has been defined for '%s'.", dmHashReverseSafe64(url));  // TODO: get the url of the co from the stack and display it here
+            lua_pushnumber(L, 0); // better than an empty hash
```

Comment:
The function is documented to return a hash, and a number is not a hash.
As mentioned, I believe I was wrong to comment on the error handling here.

I suggest going back to returning a dmhash_t from GetCollisionGroup(), and simply pushing that as the result.

(I think not having a collision group on a collision object should be a build error)

Reply:
@JCash , 

1. I assume
> (I think not having a collision group on a collision object should be a build error)
is out the scope of this issue.

2. I keep the dmLogWarning.

Reply:
UPDATE: as of the latest discussion on discord, the warning will be removed.